include/simdjson/numberparsing.h

```diff
@@ -562,9 +572,33 @@ static really_inline bool parse_number(const uint8_t *const buf, ParsedJson &pj,
       // we start anew, going slowly!!!
       return parse_float(buf, pj, offset, found_minus);
     }
-    double factor = power_of_ten[power_index];
-    factor = negative ? -factor : factor;
-    double d = i * factor;
+    double d = 0;
+    if (i != 0) {
+      components c = power_of_ten_components[power_index];
+      uint64_t factor_mantissa = c.mantissa;
+      int lz = leading_zeroes(i);
+      i <<= lz;
+      __uint128_t large_mantissa = (__uint128_t)i * factor_mantissa;
```

Comment:
128-bit support is an open question... rapidjson, for example, has a fallback when it detects that it's not possible (see their bigint.h file). it seems pretty common to support it but idk

Reply:
We need to support Visual Studio...

Reply:
ok that's fine, we can just compute this without it (3 multiplications and a few adds and shifts)